\documentclass[11pt]{article}

\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{graphicx}
\usepackage{subcaption}
\usepackage[colorinlistoftodos]{todonotes}
\usepackage[colorlinks=true, allcolors=blue]{hyperref}
% \usepackage{showlabels}
\usepackage{palatino}
\usepackage{hyphenat}
\usepackage{thm-restate}
\usepackage{natbib}
\usepackage{setspace}
\onehalfspacing
\usepackage{authblk}

% Line numbers and some weirdness
\usepackage{lineno}
\linenumbers
\newcommand*\patchAmsMathEnvironmentForLineno[1]{%
  \expandafter\let\csname old#1\expandafter\endcsname\csname #1\endcsname
  \expandafter\let\csname oldend#1\expandafter\endcsname\csname end#1\endcsname
  \renewenvironment{#1}%
     {\linenomath\csname old#1\endcsname}%
     {\csname oldend#1\endcsname\endlinenomath}}%
\newcommand*\patchBothAmsMathEnvironmentsForLineno[1]{%
  \patchAmsMathEnvironmentForLineno{#1}%
  \patchAmsMathEnvironmentForLineno{#1*}}%
\AtBeginDocument{%
\patchBothAmsMathEnvironmentsForLineno{equation}%
\patchBothAmsMathEnvironmentsForLineno{align}%
\patchBothAmsMathEnvironmentsForLineno{flalign}%
\patchBothAmsMathEnvironmentsForLineno{alignat}%
\patchBothAmsMathEnvironmentsForLineno{gather}%
\patchBothAmsMathEnvironmentsForLineno{multline}%
}

% commented for arXiv submission
%\usepackage{fancyhdr}
% \fancyhead[R]{ submission intended as Article in Discoveries section of \emph{MBE} }


\graphicspath{ {figures/} }

% http://bytesizebio.net/2013/03/11/adding-supplementary-tables-and-figures-in-latex/
\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }

% Commands
\newtheorem{theorem}{Theorem}
\newtheorem{lemma}[theorem]{Lemma}
\newtheorem*{definition}{Definition}

\newcommand{\alphabet}{\mathcal{A}}
\newcommand{\fullAlignment}{\mathbf{Y}}
\newcommand{\alignmentColumn}{\mathbf{y}}
\newcommand{\alignmentColumnRV}{Y}
\newcommand{\siteSplit}{\tilde{y}}
\newcommand{\siteSplitSet}{\mathcal{Y}}
\newcommand{\fullAncestralStates}{\mathbf{H}}
\newcommand{\ancestralStateColumn}{\mathbf{h}}
\newcommand{\ancestralStateColumnRV}{H}
\newcommand{\ancestralSplit}{\tilde{h}}
\newcommand{\ancestralSplitSet}{\mathcal{H}}
\newcommand{\ancestralSplitPartition}{\eta}
\newcommand{\fullAncestralSplitPartitions}{\boldsymbol\eta}

\newcommand{\patternToSplit}{\psi}
\newcommand{\ancestralToSplit}{\xi}

\newcommand{\siteSplitRV}{\Psi}
\newcommand{\ancestralSplitRV}{\Xi}

\newcommand{\nCols}{n}
\newcommand{\nSiteRows}{m}
\newcommand{\nAncestralStateRows}{p}
\newcommand{\nSiteSplits}{q}
\newcommand{\nAncestralSplits}{r}

\newcommand{\shannonDivergence}{D}

\DeclareMathOperator*{\argmax}{argmax}

\allowdisplaybreaks

\title{Joint maximum-likelihood of phylogenies and ancestral states is not consistent}

\author[1]{David A. Shaw}
\author[1]{Frederick A. Matsen IV\thanks{Corresponding author. Email: \url{matsen@fredhutch.org}}}
\affil[1]{Computational Biology Program, Fred Hutchinson Cancer Research Center\\ Seattle, WA, USA}
\date{}

\begin{document}

\renewcommand{\arraystretch}{1.2} % because otherwise exponents get eaten by \hline

\maketitle
% commented for arXiv submission
%\thispagestyle{fancy}

\begin{abstract}
Maximum likelihood estimation in phylogenetics requires a means of handling unknown ancestral states.
Classical maximum likelihood averages over these unknown intermediate states, leading to consistent estimation of the topology and continuous model parameters.
Recently, a computationally-efficient approach was proposed to jointly maximize over these unknown states and phylogenetic parameters.
Although this method of joint maximum likelihood estimation can obtain estimates more quickly, its properties as an estimator are not yet clear.
We show that this method of jointly estimating phylogenetic parameters along with ancestral states is not consistent in general.
We find a set of parameters that generate data under a four-taxon tree for which this joint method estimates an incorrect topology in the limit of infinite-length sequences by estimating one or more branches to be zero length.
For branch length estimation on the correct topology, we outline cases where branch length estimates are consistently and heavily biased and provide extensive empirical results for this setting.
\end{abstract}

\newpage

\section*{Introduction}

Classical maximum likelihood (ML) estimation in phylogenetics operates by integrating out latent ancestral states at the internal nodes of the tree.
In a recent paper, \citet{Sagulenko2018-xl} suggest using an approximation to ML inference in which the likelihood is maximized jointly across model parameters and ancestral sequences on a fixed topology.
This is attractive from a computational perspective: such joint inference can proceed according to an iterative procedure in which ancestral sequences are first estimated and model parameters are optimized conditional on these estimates.
This latter conditional optimization is simpler and more computationally efficient than optimizing the marginal likelihood.
But is it statistically consistent?

An estimator is said to be statistically consistent if it converges to the generating model with probability one in the large-data limit; existing consistency proofs for maximum likelihood phylogenetics \citep{Allman2008-wd,Chai2011-ff,RoyChoudhury2015-ta} apply only to estimating model parameters when the ancestral sequences have been integrated out of the likelihood.
These proofs do not readily extend to include estimating ancestral states.
Moreover, examples of inconsistency arising from problems where the number of parameters increases with the amount of data \citep{Neyman1948-tt} indicate that joint inference of trees and ancestral states may not enjoy good statistical properties.
In this case those additional parameters come in the form of the states of ancestral sequences.
%page 2 section 2.1 at the top of the second column first paragraph second sentence
Although the software described in \citet{Sagulenko2018-xl} fits on a user-supplied topology and the authors explicitly warn that the approximation is for the case where ``branch lengths are short and only a minority of sites change on a given branch,'' their work motivates understanding the general properties of such joint inference.
In particular, one would like to know when this approximate technique breaks down for both topology and branch length inference, even when sequence data is ``perfect,'' i.e., is generated without sampling error according to the exact model used for inference.

In this paper, we show that the joint inference of trees and ancestral sequences is not consistent in general.
To do so, we use a binary symmetric model with data being generated on a four-taxon tree and we construct bounds on the joint objective function to demarcate a sizeable area of branch lengths in which joint inference is guaranteed to give the wrong tree in the case of perfect sequence data with an infinite number of sites by estimating one or more branch lengths to be zero.
We find similar areas through empirical means where joint inference consistently underestimates interior branch lengths when the topology is known and fixed.

\section*{Phylogenetic maximum likelihood}

Assume the binary symmetric model, namely with a character alphabet $\alphabet=\{0,1\}$ and a uniform stationary distribution \citep{Semple2003-em}.
Let $\nSiteRows$ be the number of tips of the tree, and $\nAncestralStateRows = \nSiteRows-2$ the number of internal nodes.
We observe $\nCols$ independent and identically distributed samples of character data, i.e., an alignment with $\nCols$ columns, $\fullAlignment=[\alignmentColumn_1,\ldots,\alignmentColumn_\nCols]\in\alphabet^{\nSiteRows\times\nCols}$ distributed as the random variable $\alignmentColumnRV$.
The corresponding unobserved ancestral states are $\fullAncestralStates=[\ancestralStateColumn_1,\ldots,\ancestralStateColumn_\nCols]\in\alphabet^{\nAncestralStateRows\times\nCols}$ and distributed as $\ancestralStateColumnRV$ with each $\ancestralStateColumn_i\in\alphabet^\nAncestralStateRows$.

We parameterize branches on the unique unrooted four-tip phylogenetic tree in ways known as the ``inverse Felsenstein (InvFels)'' tree (Figs.~\ref{fig:farris-fels-top}a and~\ref{fig:farris-fels-top}b) and the ``Felsenstein'' tree (Fig.~\ref{fig:farris-fels-top}c).
The ``inverse Felsenstein'' terminology comes from \citet{Swofford2001-hr}, although it is also called the ``Farris'' tree \citep{Siddall1998-hq, Felsenstein2004}.
In the standard configuration of this tree, the interior branch parameters are equal to the bottom two parameters as in Fig.~\ref{fig:farris-fels-top}a.
We use this standard configuration as our data generating process, though we do not constrain our branch parameters to be equal in considering our optimization objective function.

\begin{figure}
\centering
\begin{subfigure}{.32\linewidth}
\centering
\includegraphics[width=\textwidth]{farris_blank_generating}
\caption[short]{InvFels tree $\tau^*$}
\end{subfigure}
\begin{subfigure}{.32\linewidth}
\centering
\includegraphics[width=\textwidth]{farris_blank_general}
\caption[short]{InvFels tree $\tau_1$}
\end{subfigure}
\begin{subfigure}{.32\linewidth}
\centering
\includegraphics[width=\textwidth]{felsenstein_blank_general}
\caption[short]{Felsenstein tree $\tau_2$}
\end{subfigure}
\caption{Three four-taxon trees with fidelities as labeled.}
\label{fig:farris-fels-top}
\end{figure}

We parameterize the branches of these trees not with the standard notion of branch length in terms of number of substitutions per site, but with an alternate formulation called ``fidelity.''
The probability of a substitution on a branch with fidelity $x$ is $(1-x)/2$ while the probability of no substitution is $(1+x)/2$ where $0 \le x \le 1$.
This parameter quantifies the fidelity of transmission of the ancestral state across an edge \citep{Matsen2007-jq}.

Fidelities have useful algebraic properties.
As data becomes plentiful, we use the Hadamard transform (see \eqref{eq:hadamard_probability} in the Appendix) to compute the exact probabilities that generate each particular configuration of taxa---we call these ``generating probabilities''---and these have an especially simple form.
For a four-taxon tree, define the general branch fidelity parameter $t=\{x_1,y_1,x_2,y_2,w\}$ where fidelities are ordered in the order of the taxa with the internal branch last (Figs.~\ref{fig:farris-fels-top}b and~\ref{fig:farris-fels-top}c).
Although we use fidelities exclusively for our theoretical development, we have made our figures in terms of probabilities of substitution $p_{x} = (1-x)/2$ as they are easier to interpret.

\subsection*{Two paths to maximum likelihood}

The standard phylogenetic likelihood approach on unrooted trees under the usual assumption of independence between sites is as follows.
For a topology $\tau$ and branch fidelities $t$ the likelihood given observed ancestral states $\fullAncestralStates$ is
\begin{equation}
\label{eq:full_likelihood}
L_\nCols(\tau, t; \fullAlignment,\fullAncestralStates) = \prod_{i=1}^{\nCols} \ \Pr(\alignmentColumnRV=\alignmentColumn_i, \ancestralStateColumnRV=\ancestralStateColumn_i \mid \tau, t).
\end{equation}
The probability $\Pr(\alignmentColumnRV=\alignmentColumn_i, \ancestralStateColumnRV=\ancestralStateColumn_i \mid \tau, t)$ is a product of transition probabilities determined by $\fullAlignment$, $\fullAncestralStates$, $\tau$, and $t$ \citep{Felsenstein2004}.

The classical approach is to maximize the likelihood marginalized across ancestral states
\begin{equation}
\label{eq:marginal_likelihood}
\tilde{L}_\nCols(\tau, t; \fullAlignment) = \prod_{i=1}^{\nCols} \ \sum_{\ancestralStateColumn_i\in\alphabet^{\nAncestralStateRows}} \ \Pr(\alignmentColumnRV=\alignmentColumn_i, \ancestralStateColumnRV=\ancestralStateColumn_i \mid \tau, t)
\end{equation}
to estimate the tree $\tau$ and branch fidelities $t$.

The alternative approach \citep{Sagulenko2018-xl} does away with the marginalization and directly estimates the maximum likelihood parameters of the fully-observed likelihood in \eqref{eq:full_likelihood}.
This is known in statistics as a profile likelihood \citep{Murphy2000-ry} or a relative likelihood \citep{Goldman1990-dk}, which exists here because $\alphabet$ is a finite set:
\begin{equation}
\label{eq:profile_likelihood}
L_\nCols'(\tau, t; \fullAlignment) = \prod_{i=1}^{\nCols} \ \max_{\ancestralStateColumn_i\in\alphabet^{\nAncestralStateRows}} \ \Pr(\alignmentColumnRV=\alignmentColumn_i, \ancestralStateColumnRV=\ancestralStateColumn_i \mid \tau, t) = \max_{\fullAncestralStates\in\alphabet^{\nAncestralStateRows\times\nCols}} \ L_\nCols(\tau, t; \fullAlignment, \fullAncestralStates).
\end{equation}
We use $\hat{\fullAncestralStates}_\nCols$ to denote an estimate for $\fullAncestralStates$ obtained by maximizing \eqref{eq:profile_likelihood}, and estimate a topology and branch fidelities using this profile likelihood as
\begin{equation}
\label{eq:profile_likelihood_topology_bl}
(\hat{\tau}_\nCols, \hat{t}_\nCols) = \argmax_{\tau, t} \ L_\nCols'(\tau, t; \fullAlignment).
\end{equation}
In general, the functional form of \eqref{eq:profile_likelihood} is determined by inequalities arising from taking maxima over ancestral state splits in Table~\ref{tab:farris_likelihoods} to obtain the likelihood term for each most likely ancestral state split, these terms depending on the unknown $(\tau, t)$.
For this reason, in practice, the joint inference strategy estimates $\hat{\fullAncestralStates}_\nCols$ for a fixed $(\tau,t)$, then $(\hat{\tau}_\nCols,\hat{t}_\nCols)$ given $\hat{\fullAncestralStates}_\nCols$, maximizing each of these conditional objectives until convergence \citep{Sagulenko2018-xl}.


\section*{Inconsistency of joint inference}

We now state our results on the inconsistency of joint inference.
All proofs are deferred to the Appendix.

Assume $\fullAlignment$ is generated from the InvFels topology $\tau^*$ and with true generating branch fidelities $t^*=\{x^*, y^*, x^*, y^*, y^*\}$ (Fig.~\ref{fig:farris-fels-top}a).
Use $\ell_{\tau^*,t^*}(\tau, t)$ to denote the expected per-site log-likelihood, which can be thought of as the infinite-length sequence case
\begin{equation}
\label{eq:exp-log-lik}
\frac{1}{n}\log L_\nCols'(\tau, t; \fullAlignment) \rightarrow \ell_{\tau^*,t^*}(\tau, t).
\end{equation}
We give $\ell$ explicitly as \eqref{eq:site_pattern_profile_likelihood_mean} in the Appendix.
For a fixed $\tau$, let $\hat{t}_\nCols$ maximize the left-hand side of \eqref{eq:exp-log-lik} and $\hat{t}$ maximize the right-hand side.
We show in the Appendix that $\hat{t}_\nCols \rightarrow \hat{t}$, allowing us to focus on only the right-hand side above.

%We show that, as $\nCols\rightarrow\infty$, there exist values for $x^*$ and $y^*$ such that the value of the likelihood after maximizing using joint inference is greater for the Felsenstein topology ($\tau_2$) than for the true, generating InvFels topology ($\tau_1$).
%To do so, we construct an upper bound $C_0(x^*, y^*)$ for the likelihood given the InvFels topology as a function of $x^*$ and $y^*$ and, similarly, a lower bound $C_1(x^*, y^*)$ for the likelihood given the Felsenstein topology.
%When $C_0(x^*, y^*) < C_1(x^*, y^*)$, the likelihood in the Felsenstein case is larger than the likelihood in the InvFels case, demonstrating inconsistency (Fig.~\ref{fig:inconsistency-farris}).

Our main theorem states that given data generated on $\tau_1$ there exist true nonzero branch lengths such that the estimator $\hat{t}$ maximizing the right-hand side of \eqref{eq:exp-log-lik} has an internal branch of length zero.
\begin{restatable}{theorem}{topoInconsist}
Let $\tau^*=\tau_1$, $t^*=\{x^*, y^*, x^*, y^*, y^*\}$, and $t=\{x_1, y_1, x_2, y_2, w\}$ with $x_1, y_1, x_2, y_2, w > 0$.
There exists a set of $0 < x^*, y^* < 1$ such that the solution $\hat{t} := \{\hat{x}_1,\hat{y}_1,\hat{x}_2,\hat{y}_2,\hat{w}\}$ given by
\[
\hat{t} = \arg\max_{t} \ \ell_{\tau^*,t^*}(\tau_1, t)
\]
has the property $\hat{w}\equiv 1$.
\end{restatable}

This result implies an inconsistency as we estimate the interior branch length to be zero (i.e., interior branch fidelity is one).
As we consider different topologies $\tau_1$ and $\tau_2$ for $\hat{t}$, the incorrect topology attains a likelihood value at its maximum equal to that of the true topology in the limit.
In other words, if $w=1$ the objective functions $\ell_{\tau^*,t^*}(\tau_1, t)$ and $\ell_{\tau^*,t^*}(\tau_2, t)$ are equivalent.
We elaborate on this point in the Appendix.
The proof is through analyzing upper and lower bounds of the potential likelihood forms in Table~\ref{tab:farris_likelihoods} and the gradients of the objective function $\ell_{\tau^*,t^*}(\tau_1, t)$ with respect to $t$.

\begin{figure}
\centering
\includegraphics[width=.95\textwidth]{incons-analytic-inkscape}
\caption{
    An analytically-derived region of inconsistency in terms of probabilities of a character change along a branch for ``perfect'' data generated on the InvFels topology (Fig.~\ref{fig:farris-fels-top}) with $p_{w^*} = p_{y^*}$ (in terms of fidelities, $w^*=y^*$).
    Due to the looseness of the upper and lower bounds, the parameters in the white region do not necessarily indicate consistency, though all parameters in the shaded region result in an inconsistency.
}
\label{fig:incons-analytic}
\end{figure}

We provide the following as an intuition for the theoretical development.
For a particular site pattern, in obtaining the joint maximum likelihood function we maximize over ancestral state splits.
For the internal branch---the branch between the two internal nodes---we have a choice of $(1+w)$ or $(1-w)$ in each of our likelihood terms depending on which ancestral state split corresponds to the highest partial likelihood term.
As $(1+w) > (1-w)$, a maximization procedure tends to prefer the $(1+w)$ term, though this will not be guaranteed because the maximum depends on the values of the unknown branch parameters $t$.
Nevertheless, this tendency to include $(1+w)$ terms in the likelihood results in a positive bias of branch fidelities, i.e., estimating branch lengths to be shorter than truth.
This is apparent in the ``long $x^*$, short $y^*$'' scenario as these are the cases in which $\emptyset$ is the most likely ancestral state split if we let $x_1=x_2=x^*$ and $y_1=y_2=y^*$ (Table~\ref{tab:likelihoods}).
If we allow multifurcating trees in our inference, then we can think of this as another instance of converging to the wrong topology, as the true $y^*\neq 1$.

\subsubsection*{Empirical validation}

Direct numerical optimization confirms our theoretically-derived bounds and provides a more detailed picture compared to the conservative analytically\hyp{}derived region (Fig.~\ref{fig:incons-analytic}).
To determine how conservative, we use the L-BFGS-B method to perform joint estimation (Fig.~\ref{fig:bl-general-inconsistency}).
We see that the region of inconsistency in the general case is much larger than in the analytic case, which is unsurprising because the analytic case dealt with loose bounds and only concerned a single set of maximal ancestral state splits.
This region encompasses almost half of the branch fidelity space; given the correct topology and performing our best possible optimization, we have many situations where we will estimate the interior branch length to be zero.

\begin{figure}
\centering
\includegraphics[width=\textwidth]{empirical-estimate}
\caption{
    Numerical estimates for $\hat{p}_w=(1-\hat{w})/2$ when optimizing \eqref{eq:profile_likelihood}, where the true value for $p_w$ is $p_{y^*}$.
    Data generated as in Fig.~\ref{fig:incons-analytic}.
    The white region in the lower right highlights which values of $x^*$ and $y^*$ result in an interior branch being estimated as length zero, resulting in an inconsistency.
}
\label{fig:bl-general-inconsistency}
\end{figure}

We provide a full description of our optimization procedure in the Appendix, but briefly, we analytically reduce the general case to 81 (see Table~\ref{tab:likelihoods}) and perform maximizations for each.
For each function, we optimize the objective starting at five separate initial values to mitigate problems with small curvature of the objective surface and nonconcavity.
We compute these maxima over a lattice in steps of $10^{-2}$ for $x^*,y^*\in(0,1)$.
Our optimization code can be found at \url{https://github.com/matsengrp/joint-inf/}.

As data were generated with parameters $\{x^*, y^*, x^*, y^*, y^*\}$, the true value for $w$ is $y^*$.
Marginal inference performs as expected, where $\hat{w}$ is equal to $y^*$ regardless of the value of $x^*$ (Fig.~\ref{fig:bl-general-marginal}) when optimizing \eqref{eq:marginal_likelihood} using the same optimization code.
For joint inference, the estimates for $\hat{p}_w$ when $p_{x^*}$ and $p_{y^*}$ are both small look reasonable, with $\hat{p}_w$ increasing as $p_{y^*}$ increases, though Fig.~\ref{fig:empirical-bias} shows there is a systematic negative bias in this procedure even when the true branches are short.
To understand the quality of each fit, we report the range of the bias $\hat{p}_w-p_{y^*}$ where $p_{x^*}, p_{y^*} \le .1$.
%EM I'd suggest bringing this figure into the main text. It'd fit.
Due to the discontinuities in the fit (Fig.~\ref{fig:bl-general-inconsistency}), an overall estimate of the bias is misleading, and these short-branch cases should be the best settings for joint optimization \citep{Sagulenko2018-xl}.
For joint inference, the errors given $p_{x^*}, p_{y^*} \le .1$ range from $[-2\times 10^{-2}, 3\times 10^{-3}]$ and for marginal inference are lower than machine tolerance showing that, even in cases where joint inference is supposed to do well, it still fails to achieve a low error from truth.

\begin{figure}
\centering
\includegraphics[width=.95\textwidth]{empirical-bias}
\caption{
Bias in branch length estimation.
In regions where joint optimization should perform well with $p_{x^*}, p_{y^*} \le .1$, there is systematic bias toward shorter branch lengths.
}
\label{fig:empirical-bias}
\end{figure}

\section*{Discussion}

We have shown that jointly inferring ancestral states and phylogenetic parameters \citep{Sagulenko2018-xl} is not consistent in general.
Specifically, in the case of four-taxon trees with infinite data, we have obtained nontrivial regions of generating parameters that result in a type of topological inconsistency: the joint inference procedure estimates zero-length branches given the correct topology.
%EM Suggest "Considering this zero length branch as a multifurcating topology..."
%EM Oops, somehow I misunderstood this the first time around. You're saying that the wrong topology, with internal branch shrunk to zero, will have the same likelihood. And that we can't exclude that it could have a greater likelihood with internal branch nonzero. If you want to make those points (or something about multifurcating) they need to be more explicit. Right now it reads like we found cases where the wrong topology has larger likelihood.
%EM Seems like clarification might be justified in other places in the MS as well.
Considering a tree with a zero-length branch as a multifurcating topology, this results in the incorrect topology having a likelihood as large or greater than that of the generating topology.
These regions of inconsistency arise when the top two branches of the generating trees are ``long,'' that is, when the top branch fidelities tend to be small, and when the lower branches are ``short,'' i.e., have large fidelities.
We see that this inconsistency occurs even if some branches are short.
This expands on the empirical findings of poor estimation given long branches in \citet{Sagulenko2018-xl} (their Figures~2 and~3).
%EM I suggest flipping this around and making it stronger, so that we lead with and emphasize the differences.
%EM I've made this a little spicier: happy with the first part?
However, the problems are not just for long branches as \citet{Sagulenko2018-xl} imply: even when all branches are short there is a consistent bias, and that the bias is on the same order as the magnitude of the parameters (Fig.~\ref{fig:empirical-bias})

Joint inference of tree parameters and ancestral sequences is a type of profile likelihood, a well-studied subject in statistics \citep{Murphy2000-ry}.
Many properties regarding the performance of maximum likelihood estimates obtained using this approach are known, and many methods exist to overcome their undesirable properties, e.g., the method of sieves \citep{Geman1982}.
A potential solution in this case using the method of sieves could be to project the column-wise ancestral states into a lower-dimensional space, allowing the degrees of freedom in the ancestral state columns to grow with $\nCols$, albeit more slowly than $O(\nCols)$.
Elsewhere in statistics literature, the failure of maximum likelihood estimates to obtain consistent estimates as the number of parameters goes to infinity have been shown by the Neyman-Scott paradox \citep{Neyman1948-tt}, though parameters tending to infinity is not a necessary condition for inconsistency \citep{LeCam1990}.
Consistency proofs of standard maximum likelihood estimates of phylogeny \eqref{eq:marginal_likelihood} are recent \citep{Allman2008-wd,Chai2011-ff,RoyChoudhury2015-ta}, and no results have been obtained for profile likelihood.
We have furthered progress in understanding the limitations of this joint optimization procedure.

Previous work in phylogenetics has developed consistency counterexamples using similar four-taxon topologies to the one used here \citep{Felsenstein1978-rr}.
In this previous work, when simulating data under the Felsenstein topology $\tau_2$, as the number of observations increases, the InvFels topology $\tau_1$ becomes more likely when performing a particular estimation procedure.
We have shown cases in which, when generating from the InvFels topology, we converge to a multifurcating topology, with one or more branch lengths estimated to be zero.
Moreover, the inconsistency demonstrated by \citet{Felsenstein1978-rr} is attributed to long branch attraction, i.e., the fact that there may be multiple long branches where parallel changes are more likely than a single change along a short branch.
This is not the case here; while analytically the inconsistency occurs in the long branch attraction region of the parameter space, we see empirically that this inconsistency is present in roughly half of the entire parameter space, and occurs when the true branches generate data that more likely has no change along the interior branch.
This is especially apparent in the ``long $x^*$, short $y^*$'' case discussed previously.
Difficulties in phylogenetic estimation when generating data on the InvFels tree have been found by \citet{Siddall1998-hq}, though \citet{Swofford2001-hr} show that sequence length plays a major role in these issues.

The case of joint inference of a phylogenetic likelihood was discussed in \citet{Goldman1990-dk}.
There, Goldman provides a worked example in which estimating a topology with fixed branch lengths is equivalent to parsimony and thus not guaranteed to be consistent, though he does not discuss the inconsistency of joint inference in general.
We show cases where the incorrect topology attains an equal likelihood value at the maximum as the correct topology, and, moreover, if we know the correct topology, we show cases where branch lengths are severely biased.
Finally, just prior to his conclusion, he discusses when parsimony gives the same answer as maximum likelihood, concluding that the question is ill-posed since parsimony estimates different parameters than maximum likelihood, i.e., it assumes equal branch lengths.
We render the question well-posed: the joint inference procedure outlined here estimates the same parameters as classical maximum likelihood---topology and branch lengths---albeit implicitly estimating ancestral states as well.
We are able to provide much more detail on how large branch lengths must be for general joint inference to fail to be consistent.

While we have shown inconsistency in topology estimation and investigated the performance of branch parameter estimation, there is substantial scope for future work to make these results more precise and more general.
The techniques used to obtain upper and lower bounds for the likelihoods in the topology estimation case provide relatively loose bounds, though how loose they are remain unknown without either further analysis or verification through simulation.
Similarly, for the general case in estimating branch lengths, we were only able to provide empirical estimates for regions of overestimation, and the discontinuities we observe via numerical optimization (Fig.~\ref{fig:bl-general-inconsistency}) beg further investigation.
Empirical validation shows that the region of inconsistency is much larger than the one analytically derived.
All of these results hold only for a simple binary symmetric model on four-taxon trees, and extensive simulation is necessary to understand how these results extend to more complicated general cases.
Examples with larger trees and more realistic mutation models fall within the scope of an extension to applied models.
Further study is needed to determine the impact of these results on problems that are of interest to practitioners.
Given that many of the bounds presented here are in the form of level sets of multivariate polynomials, a more formal approach using algebraic geometric techniques may reveal more stable or interesting patterns of inconsistency; see \citet{Sturmfels2002} for a thorough treatment of solving systems of polynomial equations.
Finally, all of the material presented here concerns joint estimation under maximum likelihood, and does not pose any problem for other settings, such as joint sampling of trees and ancestral sequences in a Bayesian framework.


\section*{Acknowledgements}
We thank Richard Neher, Vladimir Minin, and Joe Felsenstein for helpful discussions.
%TODO: change up this line?
Insight from the editors and Vu Dinh greatly improved this manuscript.

This work was supported by National Institutes of Health grants R01-GM113246, R01-AI120961, U19-AI117891, and U54-GM111274 as well as National Science Foundation grants CISE-1561334 and CISE-1564137.
The research of Frederick Matsen was supported in part by a Faculty Scholar grant from the Howard Hughes Medical Institute and the Simons Foundation.

\bibliographystyle{plainnat}
\bibliography{joint_inf}

\newpage
\beginsupplement

\input{appendix}

\end{document}
